frontend/src/pages/storageClasses/StorageClassesPage.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+
+import {
+  EmptyState,
+  EmptyStateVariant,
+  EmptyStateBody,
+  PageSection,
+  Title,
+  Alert,
+  AlertActionCloseButton,
+} from '@patternfly/react-core';
+
+import { Annotation, StorageClassConfig } from '~/k8sTypes';
+import useStorageClasses from '~/concepts/k8s/useStorageClasses';
+import { ProjectObjectType, typedEmptyImage } from '~/concepts/design/utils';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import { updateStorageClassConfig } from '~/services/StorageClassService';
+
+const StorageClassesPage: React.FC = () => {
+  const [storageClasses, storageClassesLoaded, storageClassesError] = useStorageClasses();
+  const [isAlertOpen, setIsAlertOpen] = React.useState(false);
+
+  const defaultStorageClass = storageClasses.find(
+    (storageClass) =>
+      storageClass.metadata.annotations?.[Annotation.StorageClassIsDefault] === 'true',
+  );
+
+  // Open default class alert when no default class exists
+  React.useEffect(() => {
+    setIsAlertOpen(!defaultStorageClass?.metadata.name);
+  }, [defaultStorageClass?.metadata.name]);
+
+  // Add storage class config annotations automatically for all storage classes without them
+  React.useEffect(() => {
+    storageClasses.forEach((storageClass, index) => {
+      const { metadata } = storageClass;
+      const { name: storageClassName } = metadata;
+
+      if (!metadata.annotations?.[Annotation.OdhStorageClassConfig]) {
+        const isDefault = defaultStorageClass?.metadata.uid === metadata.uid;
+        const storageClassConfig: StorageClassConfig = {
+          displayName: storageClassName,
+          ...(!defaultStorageClass
+            ? {
+                isEnabled: true,
+                isDefault: index === 0,
+              }
+            : {
+                isDefault,
+                isEnabled: isDefault,
```

Comment:
we need to constrain this more. if you have existing configured storage classes, but then later on create a new one and then set it as the new default. it would try to set it as default here when we already have a odh default set

Reply:
I think addressing you other comment above should address this.